spring-boot-starter/src/main/java/io/github/tcdl/msb/autoconfigure/MsbConfigAutoConfiguration.java

```diff
@@ -0,0 +1,108 @@
+package io.github.tcdl.msb.autoconfigure;
+
+import com.typesafe.config.Config;
+import com.typesafe.config.ConfigFactory;
+import com.typesafe.config.ConfigValueFactory;
+import io.github.tcdl.msb.config.MsbConfig;
+import io.github.tcdl.msb.support.Utils;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.AutoConfigureBefore;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+@Configuration
+@AutoConfigureBefore(MsbContextAutoConfiguration.class)
+@EnableConfigurationProperties(MsbProperties.class)
+public class MsbConfigAutoConfiguration {
+
+    public static final String DEFAULT_VERSION = "1.0.0";
+    public static final String DEFAULT_APP_NAME = Utils.generateId();
+    public static final String DEFAULT_ADAPTER_FACTORY = "io.github.tcdl.msb.adapters.amqp.AmqpAdapterFactory";
+    private static final boolean DEFAULT_BROKER_DURABLE = true;
+    public static final int DEFAULT_TIMER_THREAD_POOL_SIZE = 2;
+
+    @Autowired
+    MsbProperties msbProperties;
+
+    @Bean
+    public MsbConfig config() {
+        Config config = ConfigFactory.load("reference");
+        String appName = StringUtils.isNotBlank(msbProperties.serviceDetails.name) ? msbProperties.serviceDetails.name : DEFAULT_APP_NAME;
+        config = config.withValue("msbConfig.serviceDetails.name", ConfigValueFactory.fromAnyRef(appName));
+        config = config.withValue("msbConfig.serviceDetails.instanceId", ConfigValueFactory.fromAnyRef(msbProperties.serviceDetails.instanceId));
+
+        // Service Details
+        String version = StringUtils.isNotBlank(msbProperties.serviceDetails.version) ? msbProperties.serviceDetails.version : DEFAULT_VERSION;
+        config = config.withValue("msbConfig.serviceDetails.version", ConfigValueFactory.fromAnyRef(version));
+        if (StringUtils.isNoneBlank(msbProperties.serviceDetails.hostname))
```

Comment:
In general, I don't fully understand what is the benefit of using Spring type checked config if you put it as a string in TypeSafe config afterwards. To me it looks like *.yml / *.properties file + general mechanism for overriding would do the job. Plus, Spring supports default values for such files. You wouldn't need all this isNotBlank and null checks.